Content.Server/Psionics/Abilities/PsionicRegenerationPowerSystem.cs

```diff
@@ -0,0 +1,190 @@
+            SubscribeLocalEvent<PsionicRegenerationPowerComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<PsionicRegenerationPowerComponent, ComponentShutdown>(OnShutdown);
+            SubscribeLocalEvent<PsionicRegenerationPowerComponent, PsionicRegenerationPowerActionEvent>(OnPowerUsed);
+            SubscribeLocalEvent<PsionicRegenerationPowerComponent, MobStateChangedEvent>(OnMobStateChangedEvent);
+            SubscribeLocalEvent<PsionicRegenerationPowerComponent, DispelledEvent>(OnDispelled);
+            SubscribeLocalEvent<PsionicRegenerationPowerComponent, PsionicRegenerationDoAfterEvent>(OnDoAfter);
+        }
+
+        private void OnInit(EntityUid uid, PsionicRegenerationPowerComponent component, ComponentInit args)
+        {
+            _actions.AddAction(uid, ref component.PsionicRegenerationActionEntity, component.PsionicRegenerationActionId );
+            _actions.TryGetActionData( component.PsionicRegenerationActionEntity, out var actionData );
+            if (actionData is { UseDelay: not null })
+                _actions.StartUseDelay(component.PsionicRegenerationActionEntity);
+            if (TryComp<PsionicComponent>(uid, out var psionic))
+            {
+                psionic.ActivePowers.Add(component);
+                psionic.PsychicFeedback.Add(component.RegenerationFeedback);
+                psionic.Amplification += 0.5f;
+                psionic.Dampening += 0.5f;
+            }
+        }
+
+        private void OnPowerUsed(EntityUid uid, PsionicRegenerationPowerComponent component, PsionicRegenerationPowerActionEvent args)
+        {
+            if (!TryComp<PsionicComponent>(uid, out var psionic))
+                return;
+
+            var ev = new PsionicRegenerationDoAfterEvent(_gameTiming.CurTime);
+            var doAfterArgs = new DoAfterArgs(EntityManager, uid, component.UseDelay, ev, uid);
+
+            //Prevent the power from ignoring its own cooldown
+            _actions.TryGetActionData(component.PsionicRegenerationActionEntity, out var actionData);
+            var curTime = _gameTiming.CurTime;
+            if (actionData != null && actionData.Cooldown.HasValue && actionData.Cooldown.Value.End > curTime)
+                return;
+
+            _doAfterSystem.TryStartDoAfter(doAfterArgs, out var doAfterId);
+
+            component.DoAfter = doAfterId;
+
+            _popupSystem.PopupEntity(Loc.GetString("psionic-regeneration-begin", ("entity", uid)),
+                uid,
+                // TODO: Use LoS-based Filter when one is available.
+                Filter.Pvs(uid).RemoveWhereAttachedEntity(entity => !ExamineSystemShared.InRangeUnOccluded(uid, entity, ExamineRange, null)),
+                true,
+                PopupType.Medium);
+
+            _audioSystem.PlayPvs(component.SoundUse, uid, AudioParams.Default.WithVolume(8f).WithMaxDistance(1.5f).WithRolloffFactor(3.5f));
+
+            _psionics.LogPowerUsed(uid, "psionic regeneration",
```

Comment:
just noticed these are uID's and not strings, ignore my previous comments about these.